### DISCUSSION TITLE
Error, wiki.js does not work

### DISCUSSION THREAD

**Discussion by finnie2006** [requarks/wiki#4993] [Help / Questions]
```
[finn@Finn wiki]$ sudo node server
[sudo] password for finn: 
internal/modules/cjs/loader.js:818
  throw err;
  ^

Error: Cannot find module 'nanoid'
Require stack:
- /home/finn/Documents/wiki/server/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/finn/Documents/wiki/server/index.js:7:20)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/home/finn/Documents/wiki/server/index.js' ]
}
[finn@Finn wiki]$ npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: wiki@2.0.0
npm ERR! Found: graphql@15.3.0
npm ERR! node_modules/graphql
npm ERR!   graphql@"15.3.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer graphql@"^0.10.5 || ^0.11.3 || ^0.12.0 || ^0.13.0 || ^14.0.0" from graphql-subscriptions@1.1.0
npm ERR! node_modules/graphql-subscriptions
npm ERR!   graphql-subscriptions@"1.1.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/finn/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/finn/.npm/_logs/2022-02-09T20_10_35_930Z-debug-0.log
[finn@Finn wiki]$ 
```
When i try to run it i get this,

**Comment by tigerblue77** [comment]
Seems the same as
- https://github.com/requarks/wiki/discussions/5251
- https://github.com/requarks/wiki/discussions/4939

**Reply by finnie2006** [reply]
So.. still no answer

**Reply by tigerblue77** [reply]
Hello @NGPixel, can someone take a look please ? :)

**Comment by VGerris** [comment]
first of all, don't use sudo when not needed. 
Second of all, explain what you did, what versions you used for all things.
Looks like you did an update, usually just remove the package lock file and do npm install , not sudo.
Make sure node is setup properly , as for example suggested here : 

https://stackoverflow.com/questions/33725639/npm-install-g-less-does-not-work-eacces-permission-denied/40905762#40905762

try the latest version, docker recommended:
https://docs.requarks.io/install

If you have any issues after that, I propose to make a new ticket since this is such a while ago.
I propose to close this, thank you.

**Reply by tigerblue77** [reply]
LMAO. I know this message is not addressed to me but I can't resist answering it, considering the tone used.

First of all, answer with respect to anyone.
Second of all, this Github discussion is from a long time ago only because no contributor to this GIT repository has answered it since it was created.
Third of all, please make "Issue" section available on your Github repository so that everything you're working on will be clear and everyone could participate/contribute.
Fourth of all, although the first half of your message is disrespectful as can be, thank you for giving some hints for a solution. However, I don't see "using Docker" as a solution to the problem, which, given the number of open discussions (which I cited above), seems very real.

Thanks.

**Reply by VGerris** [reply]
Hi, first of all, maybe you should consider your own 'tone' first. Think about that it takes someone's precious time to respond. My intention was to be clear and to the point about something that is obviously not written in a matter that is complete or useful for anyone to respond to. Despite that, besides pointing that out I clearly lay out what the issue might be and supply references. I don't see how that is in any way disrespectful and quite honestly it seems you are the only one thinking that - why even respond? Write the poster to be respectful towards peoples time and explain how to initiate a proper discussion or issue instead of complaining towards the helpful. They may not help next time. I'll consider your comment as meant to be helpful, I suggest you do the same. I mostly wrote because this showed up highly in Google search results while being completely unhelpful, aiming to improve it and get it closed to keep noise out of direction. It usually takes a lot of people's spare time and dedication to keep these projects going. In my reply I have assumed that people take note of rules for posting. If you have ideas on how to go through all open discussions constructively, happy to join Thanks

**Reply by tigerblue77** [reply]
Agreed.

**Comment by DavidASix** [comment]
This issue can be resolved by adding the --legacy-peer-deps flag to your npm install command:

`npm install --legacy-peer-deps`

Once the dependencies are all installed, you can run build to set up your production env

`npm run build`
